components/plan-build/bin/shared.bash

```diff
@@ -117,12 +125,13 @@ _render_metadata_LDFLAGS(){
     local metadata_file_name="LDFLAGS"
 
     local ld_lib_part=()
-    for lib in ${pkg_lib_dirs[@]}; do
+    # shellcheck disable=2154
+    for lib in "${pkg_lib_dirs[@]}"; do
         ld_lib_part+=("-L${pkg_prefix}/$lib")
     done
-    if [[ -n ${ld_lib_part} ]]; then
+    if [[ -n ${ld_lib_part[*]} ]]; then
```

Comment:
Similar to https://github.com/habitat-sh/habitat/pull/4974#discussion_r192926905